src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -60,10 +60,23 @@ public final class SunEC extends Provider {
 
     private static final long serialVersionUID = -2279741672933606418L;
 
-    /* The property 'jdk.nativeEC' is used to control enablement of the native
-     * EC implementation.
+    // flag indicating whether the full EC implementation is present
+    // (when native library is absent then fewer EC algorithms are available)
+    private static boolean useFullImplementation = true;
+
+    private static final boolean nativeCryptTrace = NativeCrypto.isTraceEnabled();
+
+
+    /* The property 'jdk.nativeECKeyGen' is used to control enablement of the native
+     * ECKeyGeneration implementation.
+     * OpenSSL 1.1.0 or above is required for EC key generation support.
+     */
+    private static final boolean useNativeECKeyGen = NativeCrypto.isAlgorithmEnabled("jdk.nativeECKeyGen", "SunEC");
+
+    /* The property 'jdk.nativeECDH' is used to control enablement of the native
+     * ECDH implementation.
      */
-    private static final boolean useNativeEC = NativeCrypto.isAlgorithmEnabled("jdk.nativeEC", "SunEC");
+    private static final boolean useNativeECDH = NativeCrypto.isAlgorithmEnabled("jdk.nativeECDH", "SunEC");
```

Comment:
The system property `jdk.nativeEC` must continue to have the same effect.

Reply:
It's fine to add new properties, but we don't want to break applications that used `-Djdk.nativeEC=false` for whatever reason.

Reply:
Yeah, sure. I fully understand. I just wanted to have a property that is more descriptive to the user. 

So, should I revert to the previous one or use the "or" option that I mentioned?

Reply:
If we wanted to use both properties, they would be combined with `&&`, not `||`, but I don't think it's worthwhile to introduce the new property.

Reply:
It does seem like we could migrate from the use of one flag named `jdk.nativeEC` to another flag named `jdk.nativeECDH` by using a ``||` between the two values like Kostas mentioned above. I would certainly like to have two flags to turn off each feature ( ECDH vs EC key generation ) and hopefully this would allow us to have a descriptive flag using a new naming convention.

Reply:
I wouldn't be opposed to a new property to allow disabling ECDH separately from EC in general, but `jdk.nativeEC` currently controls all the OpenSSL integrations and should continue to do so, which means `&&` as I said above.

However, I'm not convinced this added flexibility is worth the complication.

Reply:
I changed it so that the property `idk.nativeEC` maintains the same functionality.